### DISCUSSION TITLE
Serialize a trigger on demand

### DISCUSSION THREAD

**Discussion by ecodina** [apache/airflow#31570] [Q&A]
Hello. A question about deferrable operators:

When an operator defers itself, it stores a "serialization" of the trigger it calls on the metadatabase. This way, the trigger can be reinstated elsewhere and airflow knows the parameters to use.

Is it at all possible to update this serialization on demand from the trigger?

**Motivation**

I have a trigger that monitors SLURM jobs (the state and the log). The trigger tails the SLURM log, so it adds the new lines on real-time.

Since the operator only defers itself once, the serialization is not updated with the number of lines already read. If the trigger is reinstated elsewhere, it reads the log from the beginning.

My current workaround is using an external database. The trigger stores the # of lines already read for a job inside the "cleanup" method, but it is not very elegant.

It is not critical: I really don't care if a few lines are repeated.

**Comment by potiuk** [comment]
No not officially supported currently. If you wish, you can make a PR with adding similar feature.
